### ISSUE TITLE
UnnecessaryParenthesesCheck does not flag unnecessary parentheses in conditional expression

### ISSUE THREAD

**Issue by mahfouz72** [checkstyle/checkstyle#15263]
I have read check documentation: https://checkstyle.org/checks/coding/unnecessaryparentheses.html#UnnecessaryParentheses
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

---

```
PS D:\CS\test> javac src/Test.java
PS D:\CS\test> cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="haltOnException" value="true"/>
  <module name="TreeWalker">
    <module name="UnnecessaryParentheses">
    </module>
  </module>
</module>
PS D:\CS\test> cat src/Test.java

public class Test {
    void test(int x) {
        boolean a = (x == 3); // violation
        boolean b = (!(x == 3)); // violation

        boolean c = (x == 3) ? (x == 3) : (!(x == 3));  // expected 3 violations
    }

}
PS D:\CS\test> java  -jar checkstyle-10.17.0-all.jar -c config.xml  src/Test.java
Starting audit...
[ERROR] D:\CS\test\src\Test.java:4:21: Unnecessary parentheses around assignment right-hand side. [UnnecessaryParentheses]
[ERROR] D:\CS\test\src\Test.java:5:21: Unnecessary parentheses around assignment right-hand side. [UnnecessaryParentheses]
Audit done.
Checkstyle ends with 2 errors.


```
---

**Describe what you expect in detail.**

`UnnecessaryParentheses` should flag unnecessary parentheses in conditional expressions. We should add `Question` to the acceptable tokens or something.

---

**Comment by rnveach** [comment]
Issue looks good to me.

**Comment by mahfouz72** [comment]
please add false negative label

**Comment by romani** [comment]
we usually place "bug" label after fix (as this label is for release notes)

**Comment by MohanadKh03** [comment]
Is this still open ? 
I think I have a solution for it but not sure if precedence order would affect this solution

**Comment by romani** [comment]
@MohanadKh03 , thanks a lot for help, you are always welcome to fix any issues.

We highly recommend (but not a mandatory  requirement) for new contributors to finish one issue in. Each group "good xxxx issue" in this case contributor learning gradually how our project works, and there less likely to be failure to finish fix.

**Comment by MohanadKh03** [comment]
@romani  
I have actually reviewed some of the previous fixes in this check especially so I have a brief idea about it.
But I have some questions regarding how should the multiple violations in a single line be reported ? Like as a `MSG_EXPR` ? 
I already have an idea about how it could be fixed so if it's possible I can raise a PR and we can discuss more details about it there ?

**Comment by romani** [comment]
Better to discuss in PR, see you in PR
